javascript/javascript2/week1/danishletter.js

```diff
@@ -0,0 +1,33 @@
+//Code For Find and count the Danish letters
+const danishString = "Jeg har en blå bil";
+
+const danishString2 = "Blå grød med røde bær";
+
+function getCountOfDanishLetters(text) {
+	const textArray = text.split("");
+	const danishLetters = { total: 0, æ: 0, ø: 0, å: 0 };
+
+	for (let i = 0; i < textArray.length; i++) {
+		let character = text.charAt(i);
+		if (text[i] === "å") {
+			danishLetters.å += 1;
+			danishLetters.total += 1;
+		} else if (text[i] === "ø") {
+			danishLetters.ø += 1;
+			danishLetters.total += 1;
+		} else if (text[i] === "æ") {
+			danishLetters.æ += 1;
+			danishLetters.total += 1;
+		}
+	}
+	//to delete the letters which is having 0 count
+	for (const item in danishLetters) {
+		if (danishLetters[item] === 0) {
+			delete danishLetters[item];
```

Comment:
In general, it is not a safe practice to delete items from an array _while_ it is being iterated. 
I would suggest one of these instead: 

- Creating a _new_ array with the elements that you want
- Even better: using declarative programming... (content of the next lesson)

Reply:
okay